providers/base/bin/gateway_ping_test.py

```diff
@@ -474,52 +400,72 @@ def main(argv) -> int:
 
     args = parse_args(argv)
 
+    if args.any_cable_interface:
+        print(_("Looking for all cable interfaces..."))
+        all_ifaces = get_default_gateways().keys()
+        args.interfaces = list(filter(is_cable_interface, all_ifaces))
+
     # If given host is not pingable, override with something pingable.
-    host = get_host_to_ping(
-        interface=args.interface, verbose=args.verbose, target=args.host
-    )
-    if args.verbose:
-        print(_("Checking connectivity to {0}").format(host))
+    host = get_host_to_ping(interface=args.interfaces[0], target=args.host)
+
+    print(_("Checking connectivity to {0}").format(host))
 
     if host:
-        ping_summary = ping(
-            host, args.interface, args.count, args.deadline, args.verbose
-        )
+        ping_summary = ping(host, args.interfaces[0])
     else:
         ping_summary = {
             "received": 0,
             "cause": "Unable to find any host to ping",
         }
 
     if ping_summary["received"] == 0:
-        print(_("No Internet connection"))
+        print(_("FAIL: All packet loss."))
         if ping_summary.get("cause"):
             print("Possible cause: {}".format(ping_summary["cause"]))
         return 1
     elif ping_summary["transmitted"] != ping_summary["received"]:
-        print(
-            _("Connection established, but lost {0}% of packets").format(
-                ping_summary["pct_loss"]
-            )
-        )
-        if ping_summary["pct_loss"] > args.threshold:
-            print(
-                _(
-                    "FAIL: {0}% packet loss is higher than {1}% threshold"
-                ).format(ping_summary["pct_loss"], args.threshold)
-            )
-            return 1
-        else:
-            print(
-                _("PASS: {0}% packet loss is within {1}% threshold").format(
-                    ping_summary["pct_loss"], args.threshold
-                )
-            )
-            return 0
+        print(_("FAIL: {0}% packet loss.").format(ping_summary["pct_loss"]))
+        return 1
     else:
-        print(_("Connection to test host fully established"))
+        print(_("PASS: 0% packet loss").format(host))
         return 0
 
 
+def get_default_gateways() -> Dict[str, str]:
+    """
+    Use `route` program to find default gateways for all interfaces.
+
+    returns a dictionary in a form of {interface_name: gateway}
+    """
+    try:
+        routes = subprocess.check_output(
+            ["route", "-n"], universal_newlines=True
```

Comment:
I didn't double check what it does to be honest, but before we were defining and envvar called `env={"LANGUAGE": "C"},`, is the omission of that intentional?

Reply:
I did not remove it by accident, but I considered that we *never* strip the locale in other places, I'd rather:
* make it consistent
* **if** the need arises it'll be better to introduce like a `locale_less_check_output` that does this to all calls of `check_output`
* make it simpler

Reply:
ok I wanted to make sure, we can go back to this if troubles arises